tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -0,0 +1,134 @@
+// <copyright file="ProcessBasicCheck.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using Spectre.Console;
+
+namespace Datadog.Trace.Tools.Runner.Checks
+{
+    internal class ProcessBasicCheck
+    {
+        private enum Runtime
+        {
+            Unknown,
+            NetFx,
+            NetCore
+        }
+
+        public static bool Run(ProcessInfo process)
+        {
+            bool foundIssue = false;
+            var runtime = DetectRuntime(process);
+
+            if (runtime == Runtime.NetFx)
+            {
+                AnsiConsole.WriteLine("Target process is running with .NET Framework");
+            }
+            else if (runtime == Runtime.NetCore)
+            {
+                AnsiConsole.WriteLine("Target process is running with .NET Core");
+            }
+            else
+            {
+                Utils.WriteWarning("Failed to detect target process runtime, assuming .NET Framework");
+                runtime = Runtime.NetFx;
+            }
+
+            var modules = FindTracerModules(process);
+
+            if (modules.Profiler == null)
+            {
+                Utils.WriteWarning("Profiler is not loaded into the process");
+                foundIssue = true;
+            }
+
+            if (modules.Tracer == null)
+            {
+                Utils.WriteWarning("Tracer is not loaded into the process");
+                foundIssue = true;
+            }
+
+            if (process.EnvironmentVariables.TryGetValue("DD_DOTNET_TRACER_HOME", out var tracerHome))
+            {
+                if (!Directory.Exists(tracerHome))
+                {
+                    Utils.WriteWarning($"DD_DOTNET_TRACER_HOME is set to {tracerHome} but the directory does not exist");
+                    foundIssue = true;
+                }
+            }
+            else
+            {
+                Utils.WriteWarning("The environment variable DD_DOTNET_TRACER_HOME is not set");
+                foundIssue = true;
+            }
+
+            string corProfilerKey = runtime == Runtime.NetCore ? "CORECLR_PROFILER" : "COR_PROFILER";
+
+            process.EnvironmentVariables.TryGetValue(corProfilerKey, out var corProfiler);
+
+            if (corProfiler != Utils.PROFILERID)
+            {
+                Utils.WriteWarning($"The environment variable {corProfilerKey} should be set to {Utils.PROFILERID} (current value: {corProfiler ?? "not set"})");
```

Comment:
Adding quote marks might make it easier to spot whitespace in the env-var issues? (I've been burned by that before). _Maybe_ could add them around `Utils.PROFILERID` too, but don't want them to copy paste them into the env var?

Reply:
Good idea

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2323/commits/786cea46f436ee5d91b283db1f93ca30207707aa